### PR TITLE
[RFC] Enable script for time-sequence prediction

### PR DESCRIPTION
**Pull request by wanchaol**
Enable script for the time-sequence prediction, did bunch of hacks to make the script mode work, and couple of issues discovered while enabling the time-sequence prediction, all noted in #8452, 

Shall we merge this PR and iteratively fix those issues thereafter?

### CODE REVIEW

File: test/test_jit.py [hunk 1]
```diff
@@ -4413,33 +4413,47 @@ def forward(self, x):
         net = Net(upscale_factor=4)
         self.checkTrace(net, (torch.rand(5, 1, 64, 64),))
 
-    @unittest.skip('This needs to be re-written into a script module')
+    # @unittest.skip('This needs to be re-written into a script module')
```
Comment (jamesr66a) [inline test/test_jit.py]: If the test runs properly you can delete this decorator line altogether

File: test/test_jit.py [hunk 2]
```diff
@@ -4413,33 +4413,47 @@ def forward(self, x):
...
                 self.lstm2 = nn.LSTMCell(51, 51)
                 self.linear = nn.Linear(51, 1)
 
-            def forward(self, input, future=0):
-                outputs = []
-                h_t = torch.zeros(input.size(0), 51, dtype=torch.double)
-                c_t = torch.zeros(input.size(0), 51, dtype=torch.double)
-                h_t2 = torch.zeros(input.size(0), 51, dtype=torch.double)
-                c_t2 = torch.zeros(input.size(0), 51, dtype=torch.double)
+            def test_lstm1(self, input, hx, cx):
```
Comment (jamesr66a) [inline test/test_jit.py]: Add a comment explaining why we need these wrappers and link to the issue

File: test/test_jit.py [hunk 3]
```diff
@@ -4413,33 +4413,47 @@ def forward(self, x):
...
+            @torch.jit.script_method
+            def forward(self, input):
+                outputs = test_tensor()
+                h_t = torch.zeros((97, 51), dtype=torch.double)
+                c_t = torch.zeros((97, 51), dtype=torch.double)
+                h_t2 = torch.zeros((97, 51), dtype=torch.double)
+                c_t2 = torch.zeros((97, 51), dtype=torch.double)
+
+                output = None
+                future = 2
```
Comment (jamesr66a) [inline test/test_jit.py]: Can you still take this as an input, just not with a default argument?

File: test/test_jit.py [hunk 4]
```diff
@@ -4413,33 +4413,47 @@ def forward(self, x):
...
+
+            @torch.jit.script_method
+            def forward(self, input):
+                outputs = test_tensor()
+                h_t = torch.zeros((97, 51), dtype=torch.double)
+                c_t = torch.zeros((97, 51), dtype=torch.double)
+                h_t2 = torch.zeros((97, 51), dtype=torch.double)
+                c_t2 = torch.zeros((97, 51), dtype=torch.double)
+
+                output = None
```
Comment (jamesr66a) [inline test/test_jit.py]: Does this work? We have an issue open for not supporting None in the script. If this works we can close it

File: test/test_jit.py [hunk 5]
```diff
@@ -4413,33 +4413,47 @@ def forward(self, x):
...
+                future = 2
+
+                for input_t in input.chunk(999, dim=1):
```
Comment (jamesr66a) [inline test/test_jit.py]: Let's add a comment showing what this chunk() call should look like in the end

**Review by jamesr66a — COMMENTED** [review]
Cool, let's document the changes that needed to be made with TODOs and links to the issues

cc @zdevito

File: test/test_jit.py (hunk 4 above)
Comment (wanchaol) [reply test/test_jit.py]: It works at least in this case. I think you are talking about #8502, I actually tested this manually and it still have that error, I think we need to figure out why.

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**